gcloud/monitoring/client.py

```diff
@@ -195,6 +203,130 @@ def metric_descriptor(self, type_,
+                       enumerated in the associated :class:`MetricDescriptor`.
+
+        :rtype: :class:`~gcloud.monitoring.metric.Metric`
+        :returns: The Metric created with the passed in arguments
+        """
+        return Metric(type=type_, labels=labels)
+
+    @staticmethod
+    def resource(type_, labels):
+        """Construct a resource object.
+
+        For a list of possible MonitoredResources and their associated labels,
+        see:
+
+        https://cloud.google.com/monitoring/api/resources
+
+        For example, to create a Resource object for a GCE instance::
+
+            >>> resource = client.resource('gce_instance', labels={
+            ...     'project_id': 'my-project-id',
+            ...     'instance_id': 'my-instance-id',
+            ...     'zone': 'us-central1-f'
+            ... })
+
+        :type type_: string
+        :param type_: The monitored resource type name.
+
+        :type labels: dict
+        :param labels: A mapping from label names to values for all labels
+                       enumerated in the associated
+                       :class:`ResourceDescriptor`.
+
+        :rtype: :class:`gcloud.resource.Resource`
+        :returns: The Resource created with the passed in arguments
+        """
+        return Resource(type_, labels)
+
+    @staticmethod
+    def timeseries(metric, resource, value,
+                   start_time=None, end_time=None):
+        """Constructs a TimeSeries object for a single data point.
+
+        .. note::
+
+           While TimeSeries objects returned by the API can have
+           multiple data points when queried as aggregates, TimeSeries created
+           on the client-side can have at most one point.
+
+        For example::
+
+            >>> timeseries = client.timeseries(metric, resource, value,
+            ...                                end_time=end)
+
+        For more information, see:
+
+        https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries
+
+        :type metric: :class:`~gcloud.monitoring.metric.Metric`
+        :param metric: A metric object.
+
+        :type resource: :class:`~gcloud.monitoring.resource.Resource`
+        :param resource: A resource object.
+
+        :type value: bool, int, string, or float
+        :param value:
+            The value of the data point to create for the TimeSeries
+
+            .. note::
+
+               This parameter can be multiple types and the TypedValue of the
+               `gcloud.monitoring.TimeSeries.Point` will be mapped from the
+               type passed in. For example, a Python float will be sent to
+               the API as a doubleValue.
+
+        :type start_time: :class:`datetime.datetime`
+        :param start_time:
+            The start time for the point written to the
+            timeseries. Defaults to None, which has meaning dependent on the
+            metric_kind of the metric being written to.
```

Comment:
"Defaults to `None`. The API takes the start time to be the same as the end time if the start time is unspecified."